### ISSUE TITLE
gopls: automated issue report (crash)

### ISSUE THREAD

**Issue by Termina1** [golang/vscode-go#2831]
gopls version: v0.12.3 (go1.20.2)
gopls flags: 
update flags: proxy
extension version: 0.39.0
go version: 1.20.2
environment: Visual Studio Code darwin
initialization error: undefined
issue timestamp: Thu, 22 Jun 2023 16:03:59 GMT
restart history:
Thu, 22 Jun 2023 16:03:41 GMT: activation (enabled: true)

ATTENTION: PLEASE PROVIDE THE DETAILS REQUESTED BELOW.

**Comment by findleyr** [comment]
Thank you for the report. Is this an open-source repository?

**Comment by Termina1** [comment]
Yes, this is https://github.com/OffchainLabs/nitro/tree/7c6240e2a4c2834f09b75d925c172f1974112ae1

**Comment by findleyr** [comment]
Not reproducing immediately for me. Does this reproduce every time for you?

Which file are you opening to get this failure?

**Comment by findleyr** [comment]
Thanks. I don't see that file in the master branch. Are you on a different branch?

**Comment by Termina1** [comment]
This one https://github.com/OffchainLabs/nitro/blob/master/arbnode/node.go

**Comment by findleyr** [comment]
Sorry, I'm having a tough time getting a valid go build in that project.

In the meantime, could you run this command and let me know if it outputs anything?
```
go run golang.org/x/tools/go/packages/gopackages -json -test -deps ./... | grep '""'
```

**Comment by findleyr** [comment]
Hrm, I do have a valid Go build now, and yet still don't reproduce. Could you also share your settings.json?

**Comment by adonovan** [comment]
I can reproduce the issue locally, just from cloning the repo and running `gopls check ./arbnode/node.go`. As I suspected, there are Metadata records being created with an empty Name field, plus an error. Specifically, for the github.com/offchainlabs/nitro/solgen/go/precompilesgen package:

util/headerreader/header_reader.go:21:2: github.com/VictoriaMetrics/fastcache@v1.6.0 (replaced by ./fastcache): reading fastcache/go.mod: open /Users/adonovan/w/nitro/fastcache/go.mod: no such file or directory

More generally, `go list` for this repo produces a bunch of errors and incomplete records that gopls appears not to be sanitizing sufficiently. (I realize this repo comes with additional instructions before one is expected to be able to run `go build`, but it shouldn't crash gopls.)
```
$ go list -deps -json ./arbnode/
{
        "Dir": "/Users/adonovan/w/nitro/util/headerreader",
        "ImportPath": "github.com/offchainlabs/nitro/util/headerreader",
 ...
        "Imports": [
                ...
                "github.com/offchainlabs/nitro/solgen/go/precompilesgen",
                ...
        ],
}
...
{
        "ImportPath": "github.com/offchainlabs/nitro/solgen/go/precompilesgen",
        "DepOnly": true,
        "Incomplete": true,
        "Stale": true,
        "StaleReason": "build ID mismatch",
        "Error": {
                "ImportStack": [
                        "github.com/offchainlabs/nitro/arbnode",
                        "github.com/offchainlabs/nitro/arbnode/dataposter",
                        "github.com/offchainlabs/nitro/util/headerreader"
                ],
                "Pos": "util/headerreader/header_reader.go:21:2",
                "Err": "github.com/VictoriaMetrics/fastcache@v1.6.0 (replaced by ./fastcache): reading fastcache/go.mod: open /Users/adonovan/w/nitro/fastcache/go.mod: no such file or directory"
        }
}
```

**Comment by findleyr** [comment]
Aha! Now I can reproduce as well. We're working on a fix.
